plugin/storage/es/esRollover.py

```diff
@@ -6,43 +6,67 @@
+        print('USAGE: [INDEX_PREFIX=(default "")] [ARCHIVE=(default false)] [SHARDS=(default {})] [REPLICAS=(default {})] [CONDITIONS=(default {})] [UNIT=(default {})] [UNIT_COUNT=(default {})] {} ACTION HOSTNAME[:PORT]'.format(SHARDS, REPLICAS, ROLLBACK_CONDITIONS, UNIT, UNIT_COUNT, sys.argv[0]))
         print('ACTION ... one of:')
         print('\tinit - creates archive index and aliases')
         print('\trollover - rollover to new write index')
         print('\tlookback - removes old indices from read alias')
         print('HOSTNAME ... specifies which ElasticSearch hosts to search and delete indices from.')
         print('INDEX_PREFIX ... specifies index prefix.')
+        print('init configuration:')
+        print('\tSHARDS ...  the number of shards per index in ElasticSearch (default 5)')
+        print('\tREPLICAS ... the number of replicas per index in ElasticSearch (default 1)')
         print('rollover configuration:')
-        print('\tCONDITIONS ... conditions used to rollover to a new write index e.g. \'{"max_age": "30d"}\'')
+        print('\tCONDITIONS ... conditions used to rollover to a new write index e.g. \'{"max_age": "7d"}\'')
         print('lookback configuration:')
         print('\tUNIT ... used with lookback to remove indices from read alias e.g. ..., days, weeks, months, years')
         print('\tUNIT_COUNT ... count of UNITs')
         sys.exit(1)
 
-    # TODO add rollover for main indices https://github.com/jaegertracing/jaeger/issues/1242
-    if not str2bool(os.getenv('ARCHIVE', 'false')):
-        print('Rollover for main indices is not supported')
-        sys.exit(1)
-
     client = elasticsearch.Elasticsearch(sys.argv[2:])
     prefix = os.getenv('INDEX_PREFIX', '')
     if prefix != '':
         prefix += '-'
-    write_alias = prefix + ARCHIVE_INDEX + '-write'
-    read_alias = prefix + ARCHIVE_INDEX + '-read'
 
     action = sys.argv[1]
+
+    if str2bool(os.getenv('ARCHIVE', 'false')):
+        write_alias = prefix + ARCHIVE_INDEX + '-write'
+        read_alias = prefix + ARCHIVE_INDEX + '-read'
+        perform_action(action, client, write_alias, read_alias, 'jaeger-span-archive')
+    else:
+        write_alias = prefix + 'jaeger-span-write'
+        read_alias = prefix + 'jaeger-span-read'
+        perform_action(action, client, write_alias, read_alias, 'jaeger-span', True)
+        write_alias = prefix + 'jaeger-service-write'
+        read_alias = prefix + 'jaeger-service-read'
+        perform_action(action, client, write_alias, read_alias, 'jaeger-service', False)
+
+
+def perform_action(action, client, write_alias, read_alias, index_to_rollover, create_template):
     if action == 'init':
-        index = prefix + ARCHIVE_INDEX + '-000001'
+        if create_template:
```

Comment:
See how the func is called, it would create templates two times in some cases.

Reply:
I can only see the three calls to the function - each with different index - so each call could pass the mapping file name. However the first call, for `span-archive` doesn't seem to create the index template - is that correct?

Reply:
The template should be created for all indices. I forgot to add the param to archive branch.

Reply:
This still needs to be done? Still think it would be better to just pass the mapping file name instead of the boolean, so each call to `perform_action` would just create the one index.